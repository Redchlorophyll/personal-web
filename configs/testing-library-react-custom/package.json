{
  "name": "testing-library-react-custom",
  "version": "0.0.0",
  "main": "./src/index.tsx",
  "types": "./src/index.tsx",
  "license": "MIT",
  "dependencies": {
    "@reduxjs/toolkit": "^1.9.5",
    "redux-setup-custom": "*"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.19.4",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.18.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/react": "^18.0.22",
    "@types/react-dom": "^18.0.7",
    "eslint": "^7.32.0",
    "eslint-config-custom": "*",
    "tsconfig": "*",
    "typescript": "^4.5.2",
    "react-intersection-observer": "^9.4.0"
  }
}
