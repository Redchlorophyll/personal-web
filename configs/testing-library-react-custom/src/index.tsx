// export * as testingLibraryDom from '@testing-library/dom';
export * from '@testing-library/react';
export * as jestDom from '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
export { userEvent };
export * from 'react-intersection-observer';

export * from './render/render-with-providers';
export * from './render/providers';
