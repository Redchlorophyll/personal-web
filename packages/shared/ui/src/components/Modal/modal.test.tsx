import React from 'react';
import '@testing-library/jest-dom';
import { Modal } from './index';

import { renderWithProviders, screen } from 'testing-library-react-custom';

describe('shared - ui - Modal', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(
      <Modal title="this is modal title">this is modal content</Modal>
    );

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('it should render properly', () => {
    renderWithProviders(
      <Modal title="this is modal title">this is modal content</Modal>
    );
    const target = screen.getByText('this is modal title');
    expect(target).toBeInTheDocument();
  });
});
