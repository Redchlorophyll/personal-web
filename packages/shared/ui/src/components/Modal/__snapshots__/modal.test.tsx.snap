// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - Modal should render properly 1`] = `
<body
  class="overflow-hidden"
>
  <div>
    <div
      class="w-full h-[100vh] fixed top-0 left-0 z-30 bg-black-900 opacity-25"
    />
    <div
      class="w-full h-[100vh] absolute z-30 flex justify-center align-baseline top-0 left-0"
    >
      <div
        class="w-[616px] min-h-[503px] bg-black-100 dark:bg-dark-layout top-1/2 fixed -translate-y-1/2 p-[25px_30px] rounded-2xl text"
      >
        <div
          class="pb-5"
        >
          <span
            class="text-base font-bold"
          >
            this is modal title
          </span>
        </div>
        <div>
          this is modal content
        </div>
      </div>
    </div>
  </div>
</body>
`;
